### DISCUSSION TITLE
Dialog pop-up to confirm commands before sending

### DISCUSSION THREAD

**Discussion by karmenlu** [yamcs/yamcs#940] [General]
Hello,

For operation purposes, we want the Yamcs web application to prompt a user to double-check or verify a command before submitting the command. I noticed an option to send commands to a holding queue to be released later and also a global configuration flag called `twoStageCommanding`. `twoStageCommanding` might be exactly what we want; however, we don't want all commands to be two-stage, only a subset. What is the most straightforward way to implement this feature to our Yamcs instance? 

Would a button listener, custom script, custom plugin, or wrapper web application be suggested?

`twoStageCommanding` is described on this page: https://docs.yamcs.org/yamcs-server-manual/web-interface/configuration/

**Comment by fqqb** [comment]
Maybe you want to look into something like Yamcs Studio, which allows you to prepare operator displays with buttons that could be configured to prompt for confirmation (https://github.com/yamcs/yamcs-studio). These displays could also be uploaded to Yamcs-Web.

As of now, on Yamcs Web, the only other option beyond what you already mention, is called `checkCommandClearance`, which is to be configured on the realtime processor in `etc/processor.yaml`:

```yaml
realtime:
  services:
    - ...
  config:
    checkCommandClearance: true
```

When set, this will add a new page `Commanding > Clearances` to the UI where users can be assigned a clearance level (by a supervisor with `ControlCommandClearances` system privilege). Then, commands can only be sent if the applicable user has the correct clearance level (info, watch, warning, distress, critical, severe) for the chosen command (based on it is significance). This functionality is not very well documented, because I'm not quite sure how I feel about it.
